document-generator-prosecution/src/main/java/uk/gov/companieshouse/document/generator/prosecution/UltimatumDocumentInfoBuilderProvider.java

```diff
@@ -0,0 +1,110 @@
+        // assetId = environmentReader.getMandatoryString(EnvironmentReaderKeys.ULTIMATUM_ASSET_ID);
+        // renderedDocDir =
+        // environmentReader.getMandatoryString(EnvironmentReaderKeys.ULTIMATUM_RENDERED_DOC_DIR);
+        // renderedDocFileBase =
+        // environmentReader.getMandatoryString(EnvironmentReaderKeys.ULTIMATUM_RENDERED_DOC_FILEBASE);
+        // templateName =
+        // environmentReader.getMandatoryString(EnvironmentReaderKeys.ULTIMATUM_TEMPLATE_NAME);
+        templateRegistryAddress =
+                        environmentReader.getMandatoryString(ConfigKeys.TEMPLATE_REGISTRY_ADDRESS);
+    }
+
+    public UltimatumDocumentInfoBuilder builder() {
+        return new UltimatumDocumentInfoBuilder();
+    }
+
+    /**
+     * Create a {@link DocumentInfoResponse} for a Prosecution Ultimatum.
+     *
+     */
+    public class UltimatumDocumentInfoBuilder {
+        private ProsecutionCase prosecutionCase;
+        private String renderedDocFileName;
+
+        private UltimatumDocumentInfoBuilder() {}
+
+        /**
+         * Sets the prosecution case, a source of info for building the template values.
+         * @param prosecutionCase
+         * @return
+         */
+        public UltimatumDocumentInfoBuilder prosecutionCase(ProsecutionCase prosecutionCase) {
+            this.prosecutionCase = prosecutionCase;
+            return this;
+        }
+
+        /**
+         * Sets the name of file for the rendered doc.
+         * @param renderedDocFileName
+         * @return
+         */
+        public UltimatumDocumentInfoBuilder renderedDocFileName(String renderedDocFileName) {
+            this.renderedDocFileName = renderedDocFileName;
+            return this;
+        }
+
+        public DocumentInfoResponse build() throws DocumentInfoCreationException {
+            DocumentInfoResponse documentInfoResponse = new DocumentInfoResponse();
+            UltimatumTemplateValues templateValues = createTemplateValues();
+            String templateValuesAsJson = toJson(templateValues);
+            documentInfoResponse.setData(templateValuesAsJson);
+            documentInfoResponse.setAssetId(assetId);
+            documentInfoResponse.setTemplateName(templateName);
+            documentInfoResponse.setPath(renderedDocDir + renderedDocFileName);
+            return documentInfoResponse;
+        }
+
+        private UltimatumTemplateValues createTemplateValues() {
+            String companyName = prosecutionCase.getCompanyName();
+            String companyIncorporationNumber = prosecutionCase.getCompanyIncorporationNumber();
+
+            UltimatumTemplateValues templateValues = new UltimatumTemplateValues();
+            templateValues.setCompanyName(companyName);
+            templateValues.setCompanyNumber(companyIncorporationNumber);
+            templateValues.setTemplateRegistryAddress(templateRegistryAddress);
+            templateValues.setDefendantName("David Fraud");
```

Comment:
Please remove example data before merging to develop. This could accidentally end up live.

Reply:
Will do